src/test/java/org/mule/extension/sftp/SftpDirectoryListenerPollFailureConsistencyFunctionaTestCase.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.extension.sftp;
+
+import static java.lang.Thread.sleep;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.collection.IsCollectionWithSize.hasSize;
+import static org.mule.extension.sftp.AllureConstants.SftpFeature.SFTP_EXTENSION;
+import static org.mule.tck.probe.PollingProber.check;
+import org.mule.extension.sftp.api.SftpFileAttributes;
+import org.mule.runtime.api.exception.MuleException;
+import org.mule.runtime.core.api.event.CoreEvent;
+import org.mule.runtime.core.api.processor.Processor;
+
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import io.qameta.allure.Description;
+import io.qameta.allure.Feature;
+import org.apache.logging.log4j.util.Strings;
+import org.junit.Test;
+
+@Feature(SFTP_EXTENSION)
+public class SftpDirectoryListenerPollFailureConsistencyFunctionaTestCase extends CommonSftpConnectorTestCase {
+
+  private static final String CONTENT = "File Content.";
+  private static final String FILE_NAME = "file_%s.txt";
+  private static final String INPUT_FOLDER = "input";
+  private static final int PROBER_TIMEOUT = 100000;
+  private static final int PROBER_DELAY = 2000;
+  private static final int FILE_CREATION_DELAY_MILLIS = 100;
+  private static final int NUMBER_OF_FILES = 100;
+
+  private static Set<String> FILES_PROCESSED;
+
+  @Override
+  public int getTestTimeoutSecs() {
+    return 300;
+  }
+
+  public static class TestProcessor implements Processor {
+
+    @Override
+    public CoreEvent process(CoreEvent event) throws MuleException {
+      synchronized (FILES_PROCESSED) {
+        if (FILES_PROCESSED != null) {
+          FILES_PROCESSED.add(((SftpFileAttributes) (event.getMessage().getAttributes().getValue())).getName());
+        }
+        return event;
+      }
+    }
+  }
+
+  public SftpDirectoryListenerPollFailureConsistencyFunctionaTestCase(String name, SftpTestHarness testHarness,
+                                                                      String ftpConfigFile) {
+    super(name, testHarness, ftpConfigFile);
+  }
+
+  @Override
+  protected String getConfigFile() {
+    return "sftp-directory-listener-with-skipped-config.xml";
+  }
+
+  @Override
+  protected void doSetUpBeforeMuleContextCreation() throws Exception {
+    super.doSetUpBeforeMuleContextCreation();
+    testHarness.makeDir(INPUT_FOLDER);
+
+    FILES_PROCESSED = new HashSet<>();
```

Comment:
if files_processed is static then this should be a concurrent implementation. `ConcurrentHashMap.newKeySet()` or `CopyOnWriteArraySet`